parser-typechecker/src/Unison/Codebase/Causal.hs

```diff
@@ -297,14 +136,25 @@ stepDistinctM
   => (e -> n e) -> Causal m h e -> n (Causal m h e)
 stepDistinctM f c = (`consDistinct` c) <$> f (head c)
 
+-- | Causal construction should go through here for uniformity;
+-- with an exception for `one`, which avoids an Applicative constraint.
+fromList :: (Applicative m, Hashable e) => e -> [Causal m h e] -> Causal m h e
```

Comment:
The name `fromList` isn't really suggesting how this should be used to me.
It's also a bit weird to have a `fromList` that takes an extra argument than just the list.

```suggestion
consOntoTails :: (Applicative m, Hashable e) => e -> [Causal m h e] -> Causal m h e
```